intents/time_compare.py

```diff
@@ -0,0 +1,180 @@
+    as well as the debiasing suggestions.
+    Some of the oversights considered in this intent are-
+
+    Args:
+        table: Type-pandas.dataframe
+            It has the contents of the csv file
+        metric: Type-string
+            It is the name of the column according to which grouping will be done.
+            summary operator is applied on metric. Metric could a column
+            containing strings, if we are applying count operator on it.
+        dimensions: Type-list of str
+            It is the name of column we want.
+            'compare batsman A and B according to total_runs',
+             dimension is 'batsman'. we group by dimensions.
+        date_range: Type-tuple
+            Tuple of start_date and end_date
+        date_column_name: Type-str
+            It is the name of column which contains date
+        date_format: Type-str
+            It is required by datetime.strp_time to parse the date in the format
+            Format Codes
+https://docs.python.org/3/library/datetime.html#strftime-and-strptime-behavior
+        slices: Type-List of tuples
+            Tuple represents the conditon to keep the row.
+            (column_name, filter, value)
+            column_name - is the value of the column that the
+            condition is applied upon.
+            filter - Filters enum members, ex. Filters.IN
+        time_compare_column: Type-list of different data types
+            first element denotes the column name by which we will do comparision.
+            second and third element are a tuple of start and end date range.
+            last element is date_formate in which the date of columns present.
+        summary_operator: Type-summary_operators enum members
+            It denotes the summary operator, after grouping by dimensions.
+            ex. SummaryOperators.MAX, SummaryOperators.SUM
+
+    Note-summary_operator is always applied on metric column passed,
+         and only when grouping is done
+
+    Returns:
+        The function will return the `table(a pandas dataframe object)`
+        after applying the intent on the
+        given `table(a pandas dataframe object)``
+    """
+
+    date_column_name = kwargs.get('date_column_name', 'date')
+    date_range = kwargs.get('date_range', None)
+    date_format = kwargs.get('date_format', '%d/%m/%Y')
+
+    slices = kwargs.get('slices', None)
+
+    table = aspects.apply_date_range(table, date_range,
+                                     date_column_name, 
+                                     date_format)
+
+    table = aspects.slice_table(table, slices)
+
+    required_columns = dimensions.copy()
+    required_columns.append(metric)
+
+    required_table = aspects.crop_other_columns(table, required_columns)
+    
+    """ we shall have two list so that we can combine both the table and 
+    after applying the groupby operation we can get the desired result."""
+    table_slice1 = aspects.apply_date_range(required_table.copy(), 
```

Comment:
is .copy() required_table.copy() really required? (I think it would be the same without .copy() )
If not is is just adding extra complexity.

Reply:
I was confused if we shall send the a table to the function then it might also change the initial table so I used required_table.copy()

Reply:
no it does not change the passed table.

Reply:
Resolved